### PR TITLE
GHA Composer v2

### PR DESCRIPTION
**Pull request by barryvdh**

### CODE REVIEW

**Comment by barryvdh** [comment]
Doesn't seem Composer 2 is stable enough yet. Logged an issue here: https://github.com/composer/composer/issues/8941#issuecomment-647970871

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -59,12 +59,14 @@ jobs:
       with:
         php-version: ${{ matrix.php }}
         coverage: none
+        tools: composer:v2
         
     - name: Install dependencies
       run: |
+        composer self-update --2
```
Comment (mfn) [inline .github/workflows/run-tests.yml:66]: This should be redundant as setup-php does this already.

Comment (barryvdh) [reply .github/workflows/run-tests.yml:66]: Yeah I added it because of the weird error I got, wanted to make sure it was the very latest version.

**Comment by barryvdh** [comment]
It works 🎉

**Comment by mfn** [comment]
No it doesn't :)

They didn't reopen the bug yet, but see https://github.com/composer/composer/issues/8903#issuecomment-651225567 and further comments.

It also requires a specific PHP runtime version 🙀 (7.2) and `composer.json` combination to crash

**Review by mfn — APPROVED** [review]
@barryvdh I took the liberty and rebased it (and also removed the `self-update` part, not necessary), and now it works => good to merge!

(I guess because we got rid of 5.5 )

**Pull request merged**